src/telemetry.ts

```diff
@@ -31,23 +31,24 @@ export class Telemetry {
     }
   }
 
-  async initialize() {
-    try {
-      this.api = await vscode.commands.executeCommand(
-        "ruby-lsp.getPrivateTelemetryApi"
-      );
-    } catch (_error) {
-      // Do nothing if no telemetry api is available
-    }
-  }
-
   async sendEvent(event: TelemetryEvent) {
-    if (this.api) {
-      return this.api.sendEvent(event);
+    if (await this.initialize()) {
+      return this.api!.sendEvent(event);
     }
   }
 
-  enabled(): boolean {
-    return Boolean(this.api);
+  private async initialize(): Promise<boolean> {
+    try {
+      if (!this.api) {
+        this.api = await vscode.commands.executeCommand(
+          "ruby-lsp.getPrivateTelemetryApi"
+        );
+      }
+
+      return Boolean(this.api);
+    } catch (_error) {
```

Comment:
Should we be more precise regarding the of errors we swallow here?

Reply:
Is that possible in JavaScript?

Reply:
Kind of: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#conditional_catch-blocks